test/utils.dart

```diff
@@ -146,50 +146,123 @@ Future<void> testBenchmarks({required bool useTallStyle}) async {
 void _testFile(TestFile testFile) {
   group(testFile.path, () {
     for (var formatTest in testFile.tests) {
-      // Collect all of the outputs that are specific to a single version.
-      // We'll run those tests at only that version. If there is is an output
-      // with no version specified, then we'll run it on all of the remaining
-      // supported versions.
-      var specificTestedVersions = {
-        for (var output in formatTest.outputs)
-          if (output.version case var version?) version,
-      };
-
-      // Run it for each of the expected outputs.
-      for (var output in formatTest.outputs) {
-        // If the output is version-specific, then only test on that version.
-        if (output.version case var version?) {
+      if (testFile.isTall) {
+        Map<Version, TestEntry> testedVersions;
+        switch (formatTest) {
+          case UnversionedFormatTest(:var output):
+            // This output is unversioned, so test at the lowest and highest
+            // supported versions. If it formats the same on those, we assume
+            // it does for all of the intermediate versions too.
+            testedVersions = {
+              _testedTallVersions.first: output,
+              _testedTallVersions.last: output,
+            };
+          case VersionedFormatTest(:var outputs):
+            testedVersions = _versionedTestEntries(outputs);
+        }
+
+        testedVersions.forEach((version, output) {
```

Comment:
[optional] Should we consider holding a preference for `for (MapEntry(key: final version, value: final output) in testedVersions.entries)` to match our guidelines around `Iterable.forEach(<functionLiteral>)`?

Reply:
If we do https://github.com/dart-lang/language/issues/2563, then yes. But right now, it's pretty verbose to use destructuring here. Or maybe if we put an extension on Map that returns and Iterable of `(K, V)` so you don't have to write `MapEntry`, `key:`, and `value:`.